src/vs/editor/standalone/browser/standaloneLanguages.ts

```diff
@@ -537,6 +537,23 @@ export interface CompletionList {
 	 */
 	items: CompletionItem[];
 }
+
+/**
+ * Contains additional information about the context in which
+ * [completion provider](#CompletionItemProvider.provideCompletionItems) is triggered.
+ */
+export interface CompletionContext {
```

Comment:
This also needs the following information  *manual* invocation, *24x7* IntelliSense, completing incomplete result set

Reply:
On a second thought we can also do that later, after this PR